src/test/TemplateCloner.test.ts

```diff
@@ -1,488 +1,492 @@
+    this._physicalModelId = physicalModelId;
+    this._functionalModelId = functionalModelId;
+    // this._drawingModelId = drawingModelId;
+    const equipmentCategoryId = definitionManager.tryGetSpatialCategoryId(SpatialCategoryName.Equipment)!;
+    this.context.remapElement(equipmentCategoryId, equipmentCategoryId); // map category of definition to category of instance - in this case the same
+  }
+
+  public async placeEquipmentInstance(equipmentDefinitionId: Id64String, placement: Placement3d, codeValue?: string): Promise<void> {
+    const equipmentDefinition = this.sourceDb.elements.getElement<DefinitionElement>(equipmentDefinitionId, DefinitionElement);
+    const physicalTemplateSql = "SELECT TargetECInstanceId FROM ElectricalEquipment:EquipmentDefinitionSpecifiesPhysicalRecipe WHERE SourceECInstanceId=:sourceId";
+    const physicalTemplateId = this.sourceDb.withPreparedStatement(physicalTemplateSql, (statement: ECSqlStatement) => {
+      statement.bindId("sourceId", equipmentDefinitionId);
+      return DbResult.BE_SQLITE_ROW === statement.step() ? statement.getValue(0).getId() : undefined;
+    });
+    // create the physical equipment by cloning/placing a template
+    let physicalInstanceId: Id64String | undefined;
+    if (physicalTemplateId) {
+      const idMap = await super.placeTemplate3d(physicalTemplateId, this._physicalModelId, placement);
+      if (codeValue) {
+        for (const clonedInstanceId of idMap.values()) {
+          const clonedInstance = this.targetDb.elements.tryGetElement<PhysicalElement>(clonedInstanceId, PhysicalElement);
+          if (clonedInstance && !clonedInstance.parent) { // The codeValue applies to the "lead" PhysicalElement (will have a null parent indicating that it is not a child)
+            physicalInstanceId = clonedInstance.id;
+            const code = this._definitionManager.createEquipmentCode(codeValue);
+            clonedInstance.code.spec = code.spec;
+            clonedInstance.code.scope = code.scope;
+            clonedInstance.code.value = code.value;
+            clonedInstance.update();
+            break;
+          }
+        }
+      }
+    }
+    // create the functional equipment
+    const functionalClassFullName = equipmentDefinition?.jsonProperties?.equipmentParams?.functionalClassFullName;
+    if (functionalClassFullName) {
+      const functionalInstanceId = this.targetDb.elements.insertElement({
+        classFullName: functionalClassFullName,
+        model: this._functionalModelId,
+        code: codeValue ? this._definitionManager.createFunctionalEquipmentCode(codeValue) : Code.createEmpty(),
+      });
+      if (physicalInstanceId) {
+        this.targetDb.relationships.insertInstance({
+          classFullName: PhysicalElementFulfillsFunction.classFullName,
+          sourceId: physicalInstanceId,
+          targetId: functionalInstanceId,
+        });
+      }
+    }
+    // WIP: waiting for missing @bentley/core-transformer placeTemplate2d API
```

Comment:
I assume this is still WIP?

Reply:
I actually don't think it still is, but I was planning on doing that in a separate PR

Reply:
it is not WIP, but I do plan on updating that in a separate PR, which I can do before I add a link to this to the itwinjs doc site

Reply:
okay, sounds good.